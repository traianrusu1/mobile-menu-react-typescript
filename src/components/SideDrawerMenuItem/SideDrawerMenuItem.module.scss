@import "../../appStyles/variables";

.sideDrawerMenuItem {
  box-sizing: border-box;
  height: 60px;
  padding: 20px 10px;
  border-top: 1px solid rgb(145, 145, 145);
  cursor: pointer;
}

.sideDrawerMenuItem:hover {
  background-color: $appColor3;
}
