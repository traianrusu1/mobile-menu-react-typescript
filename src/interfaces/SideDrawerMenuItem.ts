export interface SideDrawerMenuItem {
  name: string;
  link: string | null;
  children: SideDrawerMenuItem[] | null;
}
