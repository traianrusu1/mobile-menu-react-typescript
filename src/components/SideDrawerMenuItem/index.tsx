import SideDrawerMenuItem from './SideDrawerMenuItem';

export default SideDrawerMenuItem;
