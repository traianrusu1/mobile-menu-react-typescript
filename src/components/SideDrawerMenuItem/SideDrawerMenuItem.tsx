import React from 'react';
import styles from './SideDrawerMenuItem.module.scss';
import { SideDrawerMenuItem as SideDrawerMenuItemInterface } from '../../interfaces/SideDrawerMenuItem';

interface Props {
  item: SideDrawerMenuItemInterface;
}

const SideDrawerMenuItem: React.FC<Props> = ({ item }: Props) => {
  return <div className={styles.sideDrawerMenuItem}>{item.name}</div>;
};

export default SideDrawerMenuItem;
