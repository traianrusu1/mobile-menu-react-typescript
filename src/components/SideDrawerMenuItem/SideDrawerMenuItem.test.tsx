import React from 'react';
import { render, cleanup } from '@testing-library/react';

import SideDrawerMenuItem from './SideDrawerMenuItem';

afterEach(cleanup);

describe('<SideDrawerMenuItem />', () => {
  test('renders', () => {
    const { container } = render(<SideDrawerMenuItem myProp="" />);
    expect(container).toMatchSnapshot();
  });
});

