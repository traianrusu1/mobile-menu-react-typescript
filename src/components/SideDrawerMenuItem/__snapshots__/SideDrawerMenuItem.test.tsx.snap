// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideDrawerMenuItem /> renders 1`] = `
<div>
  <div
    class="sideDrawerMenuItem"
  >
    SideDrawerMenuItem
    
  </div>
</div>
`;
